File: app/src/main/java/com/example/themoviedb/person_details_screen/person_details_model/PersonDetailsModel.java
package com.example.themoviedb.person_details_screen.person_details_model;

import android.os.AsyncTask;

import com.example.themoviedb.person_details_screen.person_details_presenter.PersonDetailsPresenter;
import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.net.HttpURLConnection;
import java.net.MalformedURLException;
import java.net.URL;

public class PersonDetailsModel implements PersonDetailsModelInterface {
    PersonDetailsPresenter personDetailsPresenter;

    @Override
    public void setModel(PersonDetailsPresenter personDetailsPresenter) {
        this.personDetailsPresenter = personDetailsPresenter;

    }

    public class getPhotos extends AsyncTask<String, String, String> {

        HttpURLConnection httpURLConnection = null;
        URL url = null ;

        protected void onPreExecute() {
            super.onPreExecute();
        }

        @Override
        protected String doInBackground(String... urls) {
            BufferedReader reader = null;

            try {
                url = new URL(urls[0]);
                httpURLConnection = (HttpURLConnection) url.openConnection();
                httpURLConnection.connect();

                InputStream stream = httpURLConnection.getInputStream();
                reader = new BufferedReader(new InputStreamReader(stream));
                StringBuffer buffer = new StringBuffer();

                String line = "";
                while ((line = reader.readLine()) != null) {
                    buffer.append(line);
                }
                return buffer.toString();
            } catch (MalformedURLException e) {
                e.printStackTrace();
            } catch (IOException e) {
                e.printStackTrace();
            }
            finally {
                if (httpURLConnection != null)
                    httpURLConnection.disconnect();
                try {
                    if (reader != null)
                        reader.close();
                } catch (IOException e) {
                    e.printStackTrace();
                }
            }
            return null;
        }

        @Override
        protected void onPostExecute(String result) {
            super.onPostExecute(result);
            try {

                JSONObject jsonObject = new JSONObject(result);
                JSONArray jsonArray = jsonObject.getJSONArray("profiles");

                for (int i = 0; i< jsonArray.length(); i++) {

                    JSONObject profileResult = jsonArray.getJSONObject(i);
                    Profiles personImage = new Profiles();
                    personImage.setFile_path(profileResult.getString("file_path"));
                    personDetailsPresenter.addPersonImages(personImage);
                }

                personDetailsPresenter.setAdapter();

            } catch (JSONException e) {
                e.printStackTrace();
            }
        }
    }

    public void startFetchingImage(String s) {

        new getPhotos().execute(s);
    }
}

File: app/src/main/java/com/example/themoviedb/person_details_screen/person_details_model/PersonDetailsModel.kt
package com.example.themoviedb.person_details_screen.person_details_model

import android.os.AsyncTask

import com.example.themoviedb.person_details_screen.person_details_presenter.PersonDetailsPresenter
import org.json.JSONArray
import org.json.JSONException
import org.json.JSONObject

import java.io.BufferedReader
import java.io.IOException
import java.io.InputStream
import java.io.InputStreamReader
import java.net.HttpURLConnection
import java.net.MalformedURLException
import java.net.URL

class PersonDetailsModel : PersonDetailsModelInterface {

    //    PersonDetailsPresenter personDetailsPresenter;
    //
    //    @Override
    //    public void setModel(PersonDetailsPresenter personDetailsPresenter) {
    //        this.personDetailsPresenter = personDetailsPresenter;
    //
    //    }

    class GetPhotos(asyncResponse: AsyncFetchinImagesInterface) : AsyncTask<String, String, String>() {

        var asyncFetchinImagesInterface: AsyncFetchinImagesInterface? = null//Call back interface

        internal var httpURLConnection: HttpURLConnection? = null
        internal var url: URL? = null

        init {
            asyncFetchinImagesInterface = asyncResponse       //Assigning call back interface through constructor
        }

        override fun onPreExecute() {
            super.onPreExecute()
        }

        override fun doInBackground(vararg urls: String): String? {
            var reader: BufferedReader? = null

            try {
                url = URL(urls[0])
                httpURLConnection = url!!.openConnection() as HttpURLConnection
                httpURLConnection!!.connect()

                val stream = httpURLConnection!!.inputStream
                reader = BufferedReader(InputStreamReader(stream))
                val buffer = StringBuffer()

                var line = ""
                line = reader.readLine()
                    buffer.append(line)

                return buffer.toString()
            } catch (e: MalformedURLException) {
                e.printStackTrace()
            } catch (e: IOException) {
                e.printStackTrace()
            } finally {
                if (httpURLConnection != null)
                    httpURLConnection!!.disconnect()
                try {
                    reader?.close()
                } catch (e: IOException) {
                    e.printStackTrace()
                }

            }
            return null
        }

        override fun onPostExecute(result: String) {
            super.onPostExecute(result)
            try {

                val jsonObject = JSONObject(result)
                val jsonArray = jsonObject.getJSONArray("profiles")

                for (i in 0 until jsonArray.length()) {

                    val profileResult = jsonArray.getJSONObject(i)
                    val personImage = Profiles()
                    personImage.file_path = profileResult.getString("file_path")
                    asyncFetchinImagesInterface!!.processFinish(personImage)
                    //                    personDetailsPresenter.addPersonImages(personImage);
                }

                //                personDetailsPresenter.setAdapter();

            } catch (e: JSONException) {
                e.printStackTrace()
            }

        }
    }
}
